ostelco-ios-client/Network/AppErrors.swift

```diff
@@ -32,16 +35,54 @@ struct ApplicationErrors {
     /// - Parameters:
     ///   - error: The error to log
     ///   - userInfo: A dictionary with any additional user info to pass along. Defaults to nil.
+    ///   - file: The file where this method was called. Defaults to the direct caller.
+    ///   - line: The line where this method was called. Defaults to the direct caller.
     static func log(_ error: Error,
                     withAdditionalUserInfo userInfo: [String: Any]? = nil,
                     file: StaticString = #file,
                     line: UInt = #line) {
-        let fileName = (String(staticString: file) as NSString).lastPathComponent
         debugPrint("""
-            \(fileName) line \(line)
+            \(file.fileName) line \(line)
             - Error: \(error)
             - UserInfo: \(String(describing: userInfo))
             """)
         Crashlytics.sharedInstance().recordError(error, withAdditionalUserInfo: userInfo)
     }
+    
+    /// Throws an assertion failure at the point where it's called in non-production builds.
+    ///
+    /// - Parameters:
+    ///   - error: Any error conforming to `LocalizedError`
+    ///   - file: The file where this method was called. Defaults to the direct caller.
+    ///   - line: The line where this method was called. Defaults to the direct caller.
+    static func assertAndLog(_ error: LocalizedError,
+                             file: StaticString = #file,
+                             line: UInt = #line) {
+        assertionFailure(error.localizedDescription, file: file, line: line)
+        self.log(error, file: file, line: line)
+    }
+    
+    /// Throws an assertion failure at the point where it's called in non-production builds.
+    /// Logs a non-fatal error in production builds.
+    ///
+    /// - Parameters:
+    ///   - message: The message to include with the assertion failure
+    ///   - file: The file where this method was called. Defaults to the direct caller.
+    ///   - line: The line where this method was called. Defaults to the direct caller.
+    static func assertAndLog(_ message: String,
+                             file: StaticString = #file,
+                             line: UInt = #line) {
+        assertionFailure(message, file: file, line: line)
```

Comment:
This is the secret sauce that allows the assertion failure to show up at the proper file and line in the code: By passing the file and line of the caller, that assertion failure will actually show up at the spot of the caller. Very similar to how this works with `XCTest` assertions.